### ISSUE TITLE
Umbrella project in Elixir 1.2: Release fails when `build_path` is used

### ISSUE THREAD

**Issue by aclemmensen** [bitwalker/exrm#269]
I've just upgraded to Elixir 1.2 to take advantage of shared build location for umbrella projects, among other things. But it seems that when I use the path `../../_build` from inside a project in the umbrella exrm fails with an error from `resolve_release` saying that my package is an `unreachable_package`.

Removing the `build_path` configuration from `mix.exs` works around the issue but offers no compilation benefits, of course.

In other words:

1. Install Elixir 1.2
2. Create an umbrella project
3. Create a project inside the umbrella
4. Add `build_path: ../../_build` to `mix.exs` for the child project
5. Try `mix release` inside the child project folder

You'll only get an error message if you use `--verbosity=verbose`.

I'm using exrm version `0.18.1`.

**Comment by vadim-moz** [comment]
FWIW: you can circumvent the issue by creating absolute softlink:

```bash
if ! [ -L _build ]; then
  ln -s $(cd ../..; pwd)/_build _build
fi
```

I would also note that Elixir 1.2 recommends merged config in this situation:

```ex
  config_path: "../../config/config.ex",
```

It is not respected either. As result, only local config is translated to `sys.config`, and this cannot be easily alleviated. 

I submitted PR #270, which if accepted would solve the problem.

**Comment by ntrepid8** [comment]
This may be fixed by #274.  I had the same issue, can you pull and give it a try?

**Comment by vadim-moz** [comment]
Both #270 & #274 are merged. Can you pull and try again? No softlinks should be necessary.

**Comment by aclemmensen** [comment]
Appears to work perfectly. :)

I'm now experiencing another issue, though, with `erlware_commons` complaining about `neotoma` not being available although it was in my `deps` folder. I deleted it, thinking it would be restored with a `mix deps.get` but it wasn't restored. It's `relx` that brings in `erlware_commons`. I don't know if this is any way related to this change or another recent change (I was on version 0.18.1 before).

**Comment by aclemmensen** [comment]
This can be reproduced in a clean project:

1. Create umbrella project
2. Create normal project
3. Update `build_path`
4. Add dependency on `exrm` from github in the normal project
5. Run `mix deps.get`
6. Run `mix compile`

Then it'll fail with an error message.

**Comment by vadim-moz** [comment]
followed instructions to the dot. No errors (zipped project attached).

[test.zip](https://github.com/bitwalker/exrm/files/87512/test.zip)

**Comment by aclemmensen** [comment]
After trying this on another system I have to conclude that I can't reproduce it either. On my work PC I get the aforementioned error, also with your test project, on my laptop it works as intended. I could work around it by adding neotoma as an explicit dependency, so everything is in working order now. So I'll close this. Thanks for fixing the underlying issue, it seems to be a pretty big performance boost. :)
